include/gsl/gsl-lite.hpp

```diff
@@ -1293,7 +1296,38 @@ class not_null
     typedef T get_result_t;
 #endif
 
+#if gsl_CPP11_OR_GREATER
```

Comment:
We define `void_t<>` only if `gsl_CPP11_120`, so perhaps this section should also query for `gsl_CPP11_120`.

Reply:
gsl_CPP11_120 is _wider_ than gsl_CPP11_OR_GREATER, it includes every C++11-compatible compiler plus Visual Studio 2013. However, I do not know if VS2013 supports decltype and declval, so I think it's safer this way.

Reply:
Right, I missed that.
The tables at https://en.cppreference.com/w/cpp/compiler_support are useful to check compiler support for language/library features. It seems that `decltype` was added in VS2015, so `gsl_CPP11_OR_GREATER` is fine.

Reply:
Thanks for the table, I did not know that page!

But I think you misread it: decltype(auto) was added in 19.0 (VS2015), but "simple" decltype was added in 16.0 (VS2010), a.k.a. gsl_CPP11_100, so theoretically, we could use gsl_CPP11_120. However, declval() is not in the table, and I do not have access to VS2010 to try it out, so I still think we should stick to gsl_CPP11_OR_GREATER.

Reply:
Right again. Today isn't my day.
I don't think we should put too much energy into supporting C++11-related features on versions of Visual Studio older than VS2015.